src/Entity/FieldableEdgeEntityBase.php

```diff
@@ -536,4 +538,143 @@ public function getIterator(): \Traversable {
+    if (!isset($this->fieldDefinitions)) {
+      $this->getFieldDefinitions();
+    }
+    // Handle Field API fields.
+    if (isset($this->fieldDefinitions[$name])) {
+      // Support setting values via property objects.
+      if ($value instanceof TypedDataInterface) {
+        $value = $value->getValue();
+      }
+      // If a FieldItemList object already exists, set its value.
+      if (isset($this->fields[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED])) {
+        $this->fields[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED]->setValue($value);
+      }
+      // If not, create one.
+      else {
+        $this->getField($name)->setValue($value);
+      }
+    }
+    // The translations array is unset when cloning the entity object, we just
+    // need to restore it.
+    elseif ($name == 'translations') {
+      $this->translations = $value;
+    }
+    // Directly write non-field values.
+    else {
+      $this->values[$name] = $value;
+    }
+  }
+
+  /**
+   * Implements the magic method for isset().
+   */
+  public function __isset($name) {
+    // "Official" Field API fields are always set. For non-field properties,
+    // check the internal values.
+    return $this->hasField($name) ? TRUE : isset($this->values[$name]);
+  }
+
+  /**
+   * Implements the magic method for unset().
+   */
+  public function __unset($name) {
+    // Unsetting a field means emptying it.
+    if ($this->hasField($name)) {
+      $this->get($name)->setValue([]);
+    }
+    // For non-field properties, unset the internal value.
+    else {
+      unset($this->values[$name]);
+    }
+  }
+
+  /**
+   * An array of entity translation metadata.
+   *
+   * An associative array keyed by translation language code. Every value is an
+   * array containing the translation status and the translation object, if it has
+   * already been instantiated.
+   *
+   * @var array
+   */
+  private $translations = [];
+
+  /**
+   * Gets a non-translatable field.
+   *
+   * @return \Drupal\Core\Field\FieldItemListInterface
+   */
+  private function getField($name) {
+    // Populate $this->fields to speed-up further look-ups and to keep track of
+    // fields objects, possibly holding changes to field values.
+    if (!isset($this->fields[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED])) {
+      $definition = $this->getFieldDefinition($name);
+      if (!$definition) {
+        throw new \InvalidArgumentException("Field $name is unknown.");
+      }
+      // Non-translatable fields are always stored with
+      // LanguageInterface::LANGCODE_DEFAULT as key.
+      $value = NULL;
+      if (isset($this->values[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED])) {
+        $value = $this->values[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED];
+      }
+      $field = \Drupal::service('plugin.manager.field.field_type')->createFieldItemList($this->getTranslation(LanguageInterface::LANGCODE_NOT_SPECIFIED), $name, $value);
```

Comment:
`$this->getTranslation()` --- hm, we have that in place? :O

Reply:
Yes, coming from `\Drupal\apigee_edge\Entity\EdgeEntityBase::getTranslation()`.